src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -204,6 +195,35 @@ public function setup(): void
         $synchronizer->updateSchema($this->getSchema(), true);
     }
 
+    public function getMessageCount(): int
+    {
+        $queryBuilder = $this->createAvailableMessagesQueryBuilder()
+            ->select('COUNT(m.id) as message_count')
+            ->setMaxResults(1);
+
+        $data = $this->executeQuery($queryBuilder->getSQL(), $queryBuilder->getParameters())->fetch();
```

Comment:
Maybe you can directly return the result : 

```suggestion
        return $this->executeQuery($queryBuilder->getSQL(), $queryBuilder->getParameters())->fetchColumn();
```